craft_application/application.py

```diff
@@ -123,6 +124,21 @@ def __init__(
         # ``get_project`` to access the project.
         self.__project: models.Project | None = None
 
+        # Set the logging level to DEBUG for all craft-libraries. This is OK even if
+        # the specific application doesn't use a specific library, the call does not
+        # import the package.
+        util.setup_loggers(*self._cli_loggers)
+
+        craft_cli.emit.init(
```

Comment:
I'm not sure I'm comfortable with doing all of this in the Application's `__init__` method. Ideally, initialising an object shouldn't have side effects, and all three of these are side-effects.

Reply:
Then we need find somewhere to init the logger before that.

Reply:
Maybe move them to `run()`?